### DISCUSSION TITLE
Filter to compare only .cpp and .h files

### DISCUSSION THREAD

**Discussion by derricck** [WinMerge/winmerge#2416] [Q&A] [answered]
I'm trying to create a filter to compare only `.cpp` and `.h` files but it shouldn't compare folders which name `.vshistory`.

This is what i got:
```c++
## This is a directory/file filter template for WinMerge
name: C++ Files
desc: Filter for C++ source and header files

## This is an exclusive filter
## (it lets through only matching files)
def: exclude

f: \.(cpp|h)$  ## To compare only .cpp and .h files
d: \\(?<!\.vshistory)
```

Its comparing only `.cpp` and `.h` files but its not ignoring the `.vshistory` folder, whats wrong?

![image](https://github.com/user-attachments/assets/df30a094-2298-4464-abb7-52701d2c42b8)

**Comment by sdottaka** [comment, marked as answer]
If you change `d: \\(?<!\.vshistory)` to `d: \\(?<!\.vshistory)$` it should work as expected.

**Reply by derricck** [reply]
Why the $ is required?

**Reply by sdottaka** [reply]
If the regular expression is `\\(?<!\.vshistory)` and there is a folder path called `subfolder\.vshistory`, it will match the `\` in this folder path, so this folder path will not be ignored like .cpp and .h.
Adding `$` will prevent `subfolder\.vshistory` from being matched.

**Reply by sdottaka** [reply]
Oops, `\\(?<!\.vshistory)$` ignores .vshistory, but it seems to ignore other folders too.
This URL regex seems better:
https://stackoverflow.com/a/78920208/22955231
